packages/lib-classifier/src/helpers/createLocationCounts/createLocationCounts.spec.js

```diff
@@ -15,29 +15,53 @@ const mockLocations3 = [
   { 'audio/wav': 'http://foobar.com/audio1.wav' }
 ]
 
+const mockLocations4 = [
+  { 'image/jpeg': 'http://foobar.com/image1.jpeg' },
+  { 'application/json': 'http://foobar.com/data.json' }
+]
+
+const mockLocations5 = [
+  { 'image/tiff': 'http://foobar.com/image1.tiff' }
+]
+
 function mockSubject (locations) {
   return { locations }
 }
 
 describe('Helpers > createLocationCounts', function () {
-  it('should return an object', function () {
-    const result1 = createLocationCounts(mockSubject(mockLocations1))
-    expect(result1 !== null && typeof result1 === 'object').to.be.true()
-  })
+  describe('when the mime types are valid', function () {
+    it('should return an object', function () {
+      const result1 = createLocationCounts(mockSubject(mockLocations1))
+      expect(result1 !== null && typeof result1 === 'object').to.be.true()
+    })
 
-  it('should return a total location count', function () {
-    const result1 = createLocationCounts(mockSubject(mockLocations1))
-    const result2 = createLocationCounts(mockSubject(mockLocations2))
+    it('should return a total location count', function () {
+      const result1 = createLocationCounts(mockSubject(mockLocations1))
+      const result2 = createLocationCounts(mockSubject(mockLocations2))
 
-    expect(result1.total).to.equal(1)
-    expect(result2.total).to.equal(2)
-  })
+      expect(result1.total).to.equal(1)
+      expect(result2.total).to.equal(2)
+    })
+
+    it('should return the number of images', function () {
+      const result2 = createLocationCounts(mockSubject(mockLocations2))
+      const result3 = createLocationCounts(mockSubject(mockLocations3))
+
+      expect(result2.images).to.equal(2)
+      expect(result3.images).to.equal(2)
+    })
 
-  it('should return the number of images', function () {
-    const result2 = createLocationCounts(mockSubject(mockLocations2))
-    const result3 = createLocationCounts(mockSubject(mockLocations3))
+    it('should return the number of json files', function () {
+      const result4 = createLocationCounts(mockSubject(mockLocations4))
+
+      expect(result4.json).to.equal(1)
+    })
+  })
 
-    expect(result2.images).to.equal(2)
-    expect(result3.images).to.equal(2)
+  describe('when the mime types are invalid', function () {
```

Comment:
Is it worth having a text for a mix of valid and invalid types?

Reply:
I've added a test for this.